pkg/controller/networkpolicy/external_entity.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package networkpolicy provides NetworkPolicyController implementation to manage
+// and synchronize the Pods and Namespaces affected by Network Policies and enforce
+// their rules.
+
+package networkpolicy
+
+import (
+	"reflect"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/apis/core/v1alpha1"
+	"github.com/vmware-tanzu/antrea/pkg/apis/core/v1alpha2"
+)
+
+// externalEntityV1Alpha1ToAlpha2 converts a v1alpha1.ExternalEntity object to a
+// v1alpha2.ExternalEntity object.
+func externalEntityV1Alpha1ToAlpha2(ee *v1alpha1.ExternalEntity) *v1alpha2.ExternalEntity {
+	var ports []v1alpha2.NamedPort
+	var endpoints []v1alpha2.Endpoint
+	for i, ep := range ee.Spec.Endpoints {
+		if i == 0 {
```

Comment:
why only copy the ports from 1st endpoint?

Reply:
Isn' the whole point of this struct change under the assumption that for both dual-stack pods and ExternalEntities, the named ports in all endpoints will be the same?

Reply:
that's the assumption but the v1 spec allows this behavior. either way we should explicitly call out the behavior, i.e. only the  named ports in first endpoint will be copied over, or all named ports will now be associated with all endpoints

Reply:
Done.